sonic-net/sonic-buildimage
# [Radius] Fix BlastRADIUS vulnerability issue.

#### Why I did it
Always add message authenticator to fix BlastRADIUS vulnerability issue.

##### Work item tracking
- Microsoft ADO **(number only)**:

#### How I did it
Backport patches from pam_radius repo.

#### How to verify it
1. sudo config radius add 10.135.175.252
2. sudo config radius passkey <passkey>
3. sudo config aaa authentication login radius
4. Verify the Access Request includes Message-Authenticator attribute.

Captured packet:
<pre>
No.     Time           Source                Destination           eth src               vlan Protocol Length Info
   1480 18.886272      10.135.179.187        10.135.175.252        7c:c2:55:2f:d5:6d          RADIUS   150    Access-Request id=226

Frame 1480: 150 bytes on wire (1200 bits), 150 bytes captured (1200 bits)
Ethernet II, Src: 7c:c2:55:2f:d5:6d (7c:c2:55:2f:d5:6d), Dst: 5a:e5:a3:7b:19:25 (5a:e5:a3:7b:19:25)
Internet Protocol Version 4, Src: 10.135.179.187, Dst: 10.135.175.252
User Datagram Protocol, Src Port: 35993, Dst Port: 1812
RADIUS Protocol
    Code: Access-Request (1)
    Packet identifier: 0xe2 (226)
    Length: 108
    Authenticator: 0858a0b8c0023f5fa0fc3bb7c88e6c74
    [The response to this request is in frame 1481]
    Attribute Value Pairs
        AVP: t=Message-Authenticator(80) l=18 val=a0686f8d1b54ca764b52692b97f69a84
            Type: 80
            Length: 18
            Message-Authenticator: a0686f8d1b54ca764b52692b97f69a84
        AVP: t=User-Name(1) l=7 val=sonic
        AVP: t=User-Password(2) l=18 val=Encrypted
        AVP: t=NAS-IP-Address(4) l=6 val=127.0.0.1
        AVP: t=NAS-Identifier(32) l=7 val=sonic
        AVP: t=NAS-Port(5) l=6 val=238934
        AVP: t=NAS-Port-Type(61) l=6 val=Virtual(5)
        AVP: t=Service-Type(6) l=6 val=Authenticate-Only(8)
        AVP: t=Calling-Station-Id(31) l=14 val=10.154.122.2

</pre>


#### Which release branch to backport (provide reason below if selected)




#### Tested branch (Please provide the tested image version)




#### Description for the changelog




#### Link to config_db schema for YANG module changes


#### A picture of a cute animal (not mandatory but encouraged)
